docs/provisioners/oci_artifacts.md

```diff
@@ -0,0 +1,141 @@
+# Plain Provisioner
```

Comment:
OCI Artifacts Provisioner?

that said, this seems like abstraction bleed-through.  I would have expected provisioners to care about the content, not the source of that content.  So i'm surprised to see this being defined as a provisioner type.

I'd expect to have different "bundle retrieval" plugins (http, git, OCI artifact, traditional image) which know how to extract the file content, and then different provisioner plugins (plain, registryv1, etc) that know how to deal with the different kinds of file content we can expect to see.

I realize there's some bleed through because the type of bundle (plain, registryv1) may influence what files need to be extracted, but i'm still leery of a world where any new bundle format being introduced means we need to update all the bundle retrieval plugins (vs just introducing one new provisioner), and i think we can build a system that doesn't require that.

Reply:
> I'd expect to have different "bundle retrieval" plugins (http, git, OCI artifact, traditional image) which know how to extract the file content, and then different provisioner plugins (plain, registryv1, etc) that know how to deal with the different kinds of file content we can expect to see.

Yes, Ben is correct. The general idea is that there are sources which convert `bundle.spec.source` to a Go `fs.FS`. Thus far all of the provisioners use identical source implementations for consistency. At that point the provisioners job is focused on the contents of that `fs.FS` (i.e. plain vs. registry+v1 vs. helm vs. kustomize, etc.)

Reply:
@joelanford I did however have to do [this](https://github.com/operator-framework/rukpak/pull/609/files#diff-1231be96e91d457fa4b313a8f9421885b600f9be6cb15c7ec4aa978f0f2d6b27R61-R63) though, to allow a `metadata.yaml` file to be included among the artifacts, which I think brings back the discussion we were having [here](https://hackmd.io/nhAivsWVSJ6Jt0zniadDZA#Operator-Bundles-for-OLM-V1) (ref: comment on the words `The only additional, optional file is the olm.bundle.metadata.yaml file`, looks like I can't link specific comments from hackMD 😮‍💨 )